odl/space/cu_ntuples.py

```diff
@@ -170,10 +170,22 @@ def element(self, inp=None, data_ptr=None):
             if data_ptr is None:
```

Comment:
It seems you get confused by your own logic :-P
The very last line of the function will raise in this case.